evaluations/harness/common.py

```
@@ -0,0 +1,77 @@
+
+def read_deployment_ids(path: Path = IDS_CSV) -> list[str]:
+    """Read deployment IDs from a CSV.
+
+    Accepts either a header column named ``id`` or a plain one-ID-per-line file.
+    Blank lines and lines starting with ``#`` are ignored.
+    """
+    ids: list[str] = []
+    with path.open(newline="") as f:
+        sample = f.readline()
+        f.seek(0)
+        if "id" in sample.lower() and "," in sample or "id" == sample.strip().lower():
```

Ruff (0.16.3)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)